packages/container/libraries/strongly-typed/src/container.ts

```diff
@@ -0,0 +1,209 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unnecessary-type-parameters */
+import { Container, type interfaces } from 'inversify';
+
+type IfAny<T, TYes, TNo> = 0 extends 1 & T ? TYes : TNo;
+
+type BindingMapProperty = string | symbol;
+type BindingMap = Record<BindingMapProperty, any>;
+type MappedServiceIdentifier<T extends BindingMap> = IfAny<
+  T,
+  interfaces.ServiceIdentifier,
+  keyof T
+>;
+type ContainerBinding<
+  TBindingMap extends BindingMap,
+  TKey extends MappedServiceIdentifier<TBindingMap> = any,
+> = TKey extends keyof TBindingMap
+  ? TBindingMap[TKey]
+  : TKey extends interfaces.Newable<infer C>
+    ? C
+    : TKey extends interfaces.Abstract<infer D>
+      ? D
```

Comment:
I didn't see this in preview reviews..., sry. Ok, `interfaces.ServiceIdentifier<T>` no longer includes `interfaces.Abstract<T>`. Instead, it includes the expected type `Function`. Since `Abstract<T>` was just a trick to exploit the `any` type of a `Function`, what about this:
```suggestion
    : TKey extends Function
      ? any
```

Reply:
Okay not entirely sure I follow, but have done this anyway since it matches upstream.